sdk/metric/aggregator/maxsumcount/msc.go

```diff
@@ -96,6 +98,9 @@ func (c *Aggregator) Update(_ context.Context, number core.Number, desc *export.
 	c.current.count.AddUint64Atomic(1)
 	c.current.sum.AddNumberAtomic(kind, number)
 
+	c.first.Do(func() {
```

Comment:
I think with the above suggestion, we can avoid the `sync.Once` here--instead we'll just treat min and max the same way.

The `Max()` and `Min()` functions will have to recognize the minimum/maximum values as special cases. This means they'll now be able to detect a "no data" condition, which will be possible in certain race conditions within the SDK.

They can return an error to indicate that no data was found for Min or Max, using the existing declared errors in the export aggregator package.

Reply:
> Thanks @evantorrie. I'm glad you took this up. I had filed #319 which is related. It would be great if you'd like to take that up as well (in a second PR)! The task involves

Yes, this PR started off as an implementation for #319, but I then found the issue with negative values and the existing `Max()` implementation, so decided to try to fix that first.

Reply:
> The `Max()` and `Min()` functions will have to recognize the minimum/maximum values as special cases. This means they'll now be able to detect a "no data" condition, which will be possible in certain race conditions within the SDK.

@jmacd The issue with this is that when an `Aggregator` is first created, it does not have any notion of what type of `core.Number` it will be aggregating. As I understand it, it does not know until the first time that `agg.Update()` is called. 

If we do not set min/max via the sync.Once() and instead leave them as null-initialized values per the existing New() method, then the first period during which numbers are being aggregated may not give the correct answer. That's because `number.CompareNumber(kind,  c.current.max.AsNumberAtomic())` doesn't work correctly for negative values when c.current.max == 0 (and more obviously, a null-initialized c.current.min will never compare greater than an uint64 metric).

As I see it, possible solutions are:

1. Pass the type of number being aggregated down to each Aggregator's `New()` function. The only place an Aggregator is created in the SDK is when the AggregationSelector is called by the batcher, and the AggregationSelector knows which type of metric is going to be aggregated. Passing this to the Aggregator's New() would allow it to set the initialized "current.min/max" correctly based on the number kind.

2. Avoid the issue by stating that Min()/Max() are only valid after the first `Checkpoint()` has been taken for an aggregator.

I prefer approach 1., but I don't know whether there are existing constraints or other intended purposes where an Aggregator may need to be created without knowing the kind of metric that it will later aggregate.